### ISSUE TITLE
ACL cannot be updated in place, throws ACL_REDIS_RULE_NAME_ALREADY_EXISTS

### ISSUE THREAD

**Issue by atsui-bay1** [RedisLabs/terraform-provider-rediscloud#547]
### Terraform Version
Terraform v1.7.5

### Affected Resource(s)
Please list the resources as a list, for example:
- `rediscloud_acl_rule`

### Terraform Configuration Files
Original:
```hcl
resource "rediscloud_acl_rule" "acl" {
  name  = "dummy-acl"
  rule  = "+@all -@dangerous ~*"
}
```

Update
```hcl
resource "rediscloud_acl_rule" "acl" {
  name  = "dummy-acl"
  rule  = "+@all -@dangerous ~dummy:*"
}
```

Plan
```
  # rediscloud_acl_rule.acl will be updated in-place
  ~ resource "rediscloud_acl_rule" "acl" ***
        id   = "73696"
        name = "dummy-acl"
      ~ rule = "+@all -@dangerous ~*" -> "+@all -@dangerous ~dummy*"
    ***
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
I would expect the ACL to be updated in place, as it can be in the UI.  Note that providing a different name will update the existing ACL.

### Actual Behavior
```
│ Error: failed when updating redisRule 73696: 400 BAD_REQUEST - ACL_REDIS_RULE_NAME_ALREADY_EXISTS: ACL redis rule name has to be unique
│ 
│   with rediscloud_acl_rule.acl,
│   on redis.tf line 32, in resource "rediscloud_acl_rule" "acl":
│   32: resource "rediscloud_acl_rule" "acl" ***
│ 
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234

**Comment by JohnSharpe** [comment]
Hi @atsui-bay1, thanks for the catch!
It should be fixed in 1.8.0, please give it a spin and let us know if you still have trouble.
Cheers!

**Comment by atsui-bay1** [comment]
Hi @JohnSharpe , thanks for the fix, it works great!
